alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportGetAlertsAction.kt

```diff
@@ -115,22 +115,29 @@ class TransportGetAlertsAction @Inject constructor(
 
         if (getAlertsRequest.monitorId != null) {
             queryBuilder.filter(QueryBuilders.termQuery("monitor_id", getAlertsRequest.monitorId))
-            if (getAlertsRequest.workflowIds.isNullOrEmpty()) {
+            if (
+                getAlertsRequest.workflowIds != null && getAlertsRequest.workflowIds!!.size == 1 && getAlertsRequest.workflowIds!![0] == ""
+            ) {
                 val noWorkflowIdQuery = QueryBuilders.boolQuery()
                     .should(QueryBuilders.boolQuery().mustNot(QueryBuilders.existsQuery(Alert.WORKFLOW_ID_FIELD)))
                     .should(QueryBuilders.termsQuery(Alert.WORKFLOW_ID_FIELD, ""))
                 queryBuilder.must(noWorkflowIdQuery)
             }
         } else if (getAlertsRequest.monitorIds.isNullOrEmpty() == false) {
             queryBuilder.filter(QueryBuilders.termsQuery("monitor_id", getAlertsRequest.monitorIds))
-            if (getAlertsRequest.workflowIds.isNullOrEmpty()) {
+            if (
+                getAlertsRequest.workflowIds != null && getAlertsRequest.workflowIds!!.size == 1 && getAlertsRequest.workflowIds!![0] == ""
+            ) {
                 val noWorkflowIdQuery = QueryBuilders.boolQuery()
                     .should(QueryBuilders.boolQuery().mustNot(QueryBuilders.existsQuery(Alert.WORKFLOW_ID_FIELD)))
                     .should(QueryBuilders.termsQuery(Alert.WORKFLOW_ID_FIELD, ""))
                 queryBuilder.must(noWorkflowIdQuery)
             }
         }
-        if (getAlertsRequest.workflowIds.isNullOrEmpty() == false) {
+        if (
+            getAlertsRequest.workflowIds.isNullOrEmpty() == false &&
+            !(getAlertsRequest.workflowIds!!.size == 1 && getAlertsRequest.workflowIds!![0] == "")
```

Comment:
Nitpick: Would the `!(getAlertsRequest.workflowIds!!.size == 1` portion of this check be redundant given the `isNullOrEmpty` call?

Reply:
it would not be redundant because it is checking that the size should be exactly 1.